pr_agent/git_providers/git_provider.py

```diff
@@ -197,6 +197,12 @@ def auto_approve(self) -> bool:
     def calc_pr_statistics(self, pull_request_data: dict):
         return {}
 
+    def get_num_of_files(self):
+        try:
+            return len(self.get_diff_files())
+        except Exception as e:
+            return -1
```

Comment:
Consider implementing more detailed error logging within the `get_num_of_files` method. This could involve logging the type of exception and any relevant messages. This would aid in debugging and provide more context to the caller. [important]